### PR TITLE
feat: Support for custom `@format` types

### PR DESCRIPTION
**Pull request by denk0403**
### See #144.
**TL;DR**: Added support for more built-in `@format` types, as well as the ability to define custom string validation formats via the config file's `customJSDocFormats` property:

```json
{
  "customJSDocFormats": {
    "phone-number": "^\\d{3}-\\d{3}-\\d{4}$",
    "date": {
      "regex": "^\\d{4}-\\d{2}-\\d{2}$",
      "errorMessage": "Must be in YYYY-MM-DD format."
    }
  }
}
```

A nice side-effect of this implementation is that custom format types are actually able to override the output of built-ins. For example, one can override the "email" format type like so:
```js
// ts-to-zod.config.js
{ 
  // ...
  customJSDocFormats: {
    email: "^[A-Za-z]+@my_domain.com$",
  }
}
```
Then as a result, the generated Zod schema for properties with `@format email` will look like:
```js
{
  /**
   * @format email
   */
  email: z.string().regex(/^[A-Za-z]+@my_domain.com$/)
}
```

### What's left: 
- [X] Built-in support for format types `date-time`, `ip`, `ipv4`, & `ipv6`.
- [X] Support for defining custom format types via the config file.
- [X] Updating the README with documentation for the new features.
- [X] A little cleanup and better code coverage.

### CODE REVIEW

**Comment by fabien0102** [comment]
Very cool, I can't wait to see this PR merge! Good job 👌

**Review by fabien0102 — APPROVED** [review]
Amazing job!!! Thanks for this cool addition! I did modify the code style to fit the rest of the codebase.

This will be part of the next release! (and sorry for the very late review, I'm a totally out of my side-projects thoses days)
